test/Constraints/tuple.swift

```diff
@@ -161,15 +161,15 @@ func gcd_23700031<T>(_ a: T, b: T) {
 protocol Kingdom {
   associatedtype King
 }
-struct Victory<General> {
+struct Victory<General> { // expected-note {{'General' declared as parameter to type 'Victory'}}
   init<K: Kingdom>(_ king: K) where K.King == General {}
 }
 struct MagicKingdom<K> : Kingdom {
   typealias King = K
 }
 func magify<T>(_ t: T) -> MagicKingdom<T> { return MagicKingdom() }
 func foo(_ pair: (Int, Int)) -> Victory<(x: Int, y: Int)> {
-  return Victory(magify(pair)) // expected-error {{cannot convert return expression of type 'Victory<(Int, Int)>' to return type 'Victory<(x: Int, y: Int)>'}}
+  return Victory(magify(pair)) // expected-error {{generic parameter 'General' could not be inferred}} expected-note {{explicitly specify the generic arguments to fix this issue}}
```

Comment:
I'm not sure this is a better error message. Before, we got a (correct) type mismatch due to the subexpression type-checking fine on its own; now we complain about the parameter not being inferred (because there is conflicting contextual information). @rudkx, what do you think?